core/src/identity/ed25519.rs

```diff
@@ -100,7 +100,7 @@ pub struct PublicKey(ed25519::PublicKey);
 impl PublicKey {
     /// Verify the Ed25519 signature on a message using the public key.
     pub fn verify(&self, msg: &[u8], sig: &[u8]) -> bool {
```

Comment:
You might want to consider changing this method to return a `Result<(), ()>` so as to strategically avoid this bugclass in the future